core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -54,6 +64,10 @@ protected BaseTaskWriter(PartitionSpec spec, FileFormat format, FileAppenderFact
     this.targetFileSize = targetFileSize;
   }
 
+  public Set<CharSequence> referencedDataFiles() {
+    return referencedDataFiles;
```

Comment:
Good catch. Should this set be part of the `WriteResult` instead of separate? I think that tasks are going to need to pass the set back to the commit for validation, so adding it to the `WriteResult` seems like the right way to handle it.